src/main/main.ts

```diff
@@ -576,8 +643,8 @@ const doSplashScreenChecks = async () =>
             await sleep(250);
 
             splash.webContents.send('checking-deps');
-            await checkPythonDeps(splash);
-            await checkNvidiaSmi();
+            const hasNvidia = await checkNvidiaSmi();
```

Comment:
This check plus its side effects don't need python at all, right? If not, then we could start the promise at the top of the function and only await the result here. That way, the result is likely already ready when we need its result, and we won't have to wait.

Reply:
That is correct, we could do that